src/stock-on-hand-mobile/components/stock-on-hand-table-component.jsx

```diff
@@ -0,0 +1,103 @@
+/*
+ * This program is part of the OpenLMIS logistics management information system platform software.
+ * Copyright © 2017 VillageReach
+ *
+ * This program is free software: you can redistribute it and/or modify it under the terms
+ * of the GNU Affero General Public License as published by the Free Software Foundation, either
+ * version 3 of the License, or (at your option) any later version.
+ *  
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. 
+ * See the GNU Affero General Public License for more details. You should have received a copy of
+ * the GNU Affero General Public License along with this program. If not, see
+ * http://www.gnu.org/licenses.  For additional information contact info@OpenLMIS.org. 
+ */
+
+import React from 'react';
+import { useTable } from 'react-table';
+
+const StockOnHandTable = ({
+    columns,
+    data,
+    hiddenColumns,
+    expandedProducts,
+    ...props
+}) => {
+    const {
+        getTableProps,
+        getTableBodyProps, 
+        headerGroups, 
+        rows, 
+        prepareRow
+    } = useTable(
+        {
+            columns,
+            data,
+            hiddenColumns
+        },
+    );
+    
+    return (
+        <>
+            <table { ...getTableProps() } >
+                <thead>
+                {headerGroups.map(headerGroup => (
+                    <tr {...headerGroup.getHeaderGroupProps()} >
+                        {headerGroup.headers.map((column) => {if (!column.hideHeader) {
+                            return (
+                            <th {...column.getHeaderProps()}>{column.render('Header')}</th>
+                            );
+                        }
+                        return  <th className='hidden'/>
+                    }
+                        )}
+                    </tr>
+                ))}
+                </thead>
+                <tbody {...getTableBodyProps()}>
+                    {rows.map(row => {
+                        const productInfo = row.original.canFulfillForMe.map((product) => {
+                            return (
+                                <tr className={`row-expanded ${expandedProducts.indexOf(row.original.orderable.id) == -1 && 'hidden'}`}>
+                                    <td className='cell-expanded'>
+                                        <div>Lot Code</div>
+                                        <div>Quantity</div>
+                                    </td>
+                                    <td/>
+                                    <td className='cell-expanded'>
+                                        <div>
+                                            {product.lot ? product.lot.lotCode : 'No lot defined'}
```

Comment:
I think this line could be written in such way using optional chaining:
                                   `         {product?.lot?.lotCode ??  'No lot defined'}`